test/extensions/quic_listeners/quiche/envoy_quic_dispatcher_test.cc

```diff
@@ -0,0 +1,226 @@
+#include <openssl/evp.h>
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+
+#include "quiche/quic/core/quic_dispatcher.h"
+#include "quiche/quic/test_tools/crypto_test_utils.h"
+#include "quiche/quic/test_tools/quic_test_utils.h"
+#include "quiche/quic/platform/api/quic_text_utils.h"
+#pragma GCC diagnostic pop
+
+#include <memory>
+
+#include "extensions/quic_listeners/quiche/envoy_quic_connection_helper.h"
+#include "common/network/listen_socket_impl.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/environment.h"
+#include "test/mocks/network/mocks.h"
+#include "test/test_common/utility.h"
+#include "test/test_common/network_utility.h"
+#include "extensions/quic_listeners/quiche/platform/envoy_quic_clock.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_utils.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_dispatcher.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_fake_proof_source.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_alarm_factory.h"
+#include "server/configuration_impl.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::Invoke;
+using testing::Return;
+using testing::ReturnRef;
+
+namespace quic {
+namespace test {
+class QuicDispatcherPeer {
```

Comment:
Can we use QuicDispatcherPeer from quic/test_tools/quic_dispatcher_peer.h?

Reply:
That peer class doesn't expose time_waitlist_manager_.